app/javascript/components/widgets/numbers/sleeping-blocks.jsx

```diff
@@ -55,13 +55,18 @@ Sleeping.update = function freezeOnSleep(bodies, timeScale) {
       ) {
         body.sleepCounter += 1;
 
-        if (body.sleepCounter >= body.sleepThreshold) {
+        if (body.sleepCounter >= body.sleepThreshold && body.render.visible) {
```

Comment:
Why is `body.render.visible` necessary here? Is `sleepCounter` lost during serialization?

Reply:
The objects that are currently "sleeping" pass the check, as they still exist but are simply invisible and static. Given that they pass this check, they would be redrawn to the second canvas as well as making unnecessary calculations. Technically this could be checking for if the body is static instead, as the body being invisible implies the body is static in our usage.